netebpfext/net_ebpf_ext_xdp.c

```diff
@@ -372,13 +394,15 @@ _net_ebpf_ext_allocate_cloned_nbl(_Inout_ net_ebpf_xdp_md_t* net_xdp_ctx, uint32
 }
 
 static void
-_net_ebpf_ext_free_nbl(_Inout_ NET_BUFFER_LIST* nbl)
+_net_ebpf_ext_free_nbl(_Inout_ NET_BUFFER_LIST* nbl, bool free_data)
 {
     NET_BUFFER* net_buffer = NET_BUFFER_LIST_FIRST_NB(nbl);
     MDL* mdl_chain = NET_BUFFER_FIRST_MDL(net_buffer);
-    uint8_t* buffer = (uint8_t*)MmGetSystemAddressForMdlSafe(mdl_chain, NormalPagePriority);
-    if (buffer != NULL)
-        ExFreePool(buffer);
+    if (free_data) {
+        uint8_t* buffer = (uint8_t*)MmGetSystemAddressForMdlSafe(mdl_chain, NormalPagePriority);
+        if (buffer != NULL)
```

Comment:
missing braces

Reply:
There are a lot of if statements without braces in the code base. Is this a recent coding guideline change?

Reply:
It's not new.  
https://github.com/microsoft/ebpf-for-windows/blob/main/docs/DevelopmentGuide.md#formatting-notes has always said:
> Single-line if/else/loop blocks MUST be enclosed in braces.

Reply:
Generally, I ignore existing single line, conditional/loop blocks, but do look for them in _new_ code per guidelines in our [GettingStarted doc](https://github.com/microsoft/ebpf-for-windows/blob/main/docs/DevelopmentGuide.md#formatting-notes)

Reply:
FYI clang-format only supports enforcing it starting in clang v15 (see https://stackoverflow.com/questions/26111162/can-clang-format-add-braces-to-single-line-if-statements-etc and https://clang.llvm.org/docs/ClangFormatStyleOptions.html)

Reply:
> FYI clang-format only supports enforcing it starting in clang v15 (see https://stackoverflow.com/questions/26111162/can-clang-format-add-braces-to-single-line-if-statements-etc and https://clang.llvm.org/docs/ClangFormatStyleOptions.html)

Yep.  _I'd_ actually added the explicit note re. the brace requirement for single line conditional/loop blocks to our GettingStarted.md doc due to our inability to use clang-15 for formatting which supported this.  (I believe I'd raised this in our weekly meetings).